rest/import_test.go

```diff
@@ -698,6 +698,236 @@ func TestMigrateWithExternalRevisions(t *testing.T) {
+	assertXattrSyncMetaRevGeneration(t, bucket, mobileKey, 1)
+
+	// Verify the expiry has been preserved after the import
+	gocbBucket := bucket.(*base.CouchbaseBucketGoCB)
+	expiry, err := gocbBucket.GetExpiry(mobileKey)
+	assertNoError(t, err, "Error calling GetExpiry()")
+	assert.True(t, expiry == uint32(expiryUnixEpoch))
+
+}
+
+// Test migration of a 1.5 doc that has an expiry value.
+func TestFeedBasedMigrateWithExpiry(t *testing.T) {
+
+	SkipImportTestsIfNotEnabled(t)
+
+	rt := RestTester{
+		SyncFn: `function(doc, oldDoc) { channel(doc.channels) }`,
+		DatabaseConfig: &DbConfig{
+			ImportDocs: "continuous",
+		},
+	}
+	defer rt.Close()
+	bucket := rt.Bucket()
+
+	rt.SendAdminRequest("PUT", "/_logging", `{"Import+":true, "CRUD+":true}`)
+
+	// Write doc in SG format directly to the bucket
+	key := "TestFeedBasedMigrateWithExpiry"
+
+	// Create via the SDK with sync metadata intact
+	expirySeconds := time.Second * 30
+	testExpiry := time.Now().Add(expirySeconds)
+	bodyString := rawDocWithSyncMeta()
+	_, err := bucket.Add(key, uint32(testExpiry.Unix()), []byte(bodyString))
+	assertNoError(t, err, "Error writing doc w/ expiry")
+
+	// Wait for doc to appear on changes feed
+	// Wait until the change appears on the changes feed to ensure that it's been imported by this point
+	now := time.Now()
+	changes, err := rt.WaitForChanges(1, "/db/_changes", "", true)
+	assertNoError(t, err, "Error waiting for changes")
+	changeEntry := changes.Results[0]
+	assert.Equals(t, changeEntry.ID, key)
+	log.Printf("Saw doc on changes feed after %v", time.Since(now))
+
+	// Double-check to make sure that it's been imported by checking the Sync Metadata in the xattr
+	assertXattrSyncMetaRevGeneration(t, bucket, key, 1)
+
+	// Now get the doc expiry and validate that it has been migrated into the doc metadata
+	gocbBucket := bucket.(*base.CouchbaseBucketGoCB)
+	expiry, err := gocbBucket.GetExpiry(key)
+	assert.True(t, expiry > 0)
+	assertNoError(t, err, "Error calling GetExpiry()")
+	log.Printf("expiry: %v", expiry)
+	assert.True(t, expiry == uint32(testExpiry.Unix()))
+
+}
+
+
+// Write a doc via SDK with an expiry value.  Verify that expiry is preserved when doc is imported via on-demand
+// import (GET or WRITE)
+func TestXattrOnDemandImportPreservesExpiry(t *testing.T) {
+
+	SkipImportTestsIfNotEnabled(t)
+
+	var rt RestTester
+
+	mobileBody := make(map[string]interface{})
+	mobileBody["type"] = "mobile"
+	mobileBody["channels"] = "ABC"
+
+	triggerOnDemandViaGet := func(key string) {
+		rt.SendAdminRequest("GET", fmt.Sprintf("/db/%s", key), "")
+	}
+	triggerOnDemandViaWrite := func(key string) {
+		mobileBody["foo"] = "bar"
+		mobileBodyMarshalled, err := json.Marshal(mobileBody)
+		assertNoError(t, err, "Error marshalling body")
+		rt.SendAdminRequest("PUT", fmt.Sprintf("/db/%s", key), string(mobileBodyMarshalled))
+	}
+
+	testCases := []func(string){
+		triggerOnDemandViaGet,
+		triggerOnDemandViaWrite,
+	}
+
```

Comment:
Switch to something like:

testCase struct
{
  name string
  callback func()
}